Tutorials/WinForms/Classic/bind-the-data-grid-to-large-data-source.md

```diff
@@ -0,0 +1,294 @@
+    * Name = Freight, Property = [Freight]
+* Delete the **CustomersBindingSource** component
+* Change the **CustomersGridControl** name to **OrdersGridControl** and do the same for **CustomersGridView**
+* Set the **OrdersGridControl.DataSource** property to **OrdersInstantFeedbackView**
+* Select the **OrdersGridControl** control on the design surface and click the smart tag button in the top-right corner to open the menu. Click the **Run Designer** menu item
+* Select the **Columns** item in the navigation panel and click the **Retrieve Fields** button in the toolbar
+* Select the **Layout** item in the navigation panel, drag the **Oid** grid column down and release it when the cross icon appears. The hidden **Oid** column will be used later to obtain object identifiers. Click the **Apply** button, close the designer, and save changes
+* Open the code editor and change the code as follows:
+    <details>
+    <summary>Click to expand</summary>
+
+    ```csharp
+    using DevExpress.Xpo;
+    using DevExpress.XtraBars;
+    using DevExpress.XtraBars.Ribbon;
+    using DevExpress.XtraGrid.Views.Grid;
+    using DxSample.Forms;
+    using System;
+
+    namespace DxSample {
+        public partial class OrdersListForm : RibbonForm {
+            public OrdersListForm() {
+                InitializeComponent();
+            }
+            private void CustomersGridView_RowClick(object sender, RowClickEventArgs e) {
+                if (e.Clicks == 2) {
+                    e.Handled = true;
+                    int orderID = (int)OrdersGridView.GetRowCellValue(e.RowHandle, colOid);
+                    ShowEditForm(orderID);
+                }
+            }
+
+            private void ShowEditForm(int? orderID) {
+                using (EditOrderForm form = new EditOrderForm(orderID)) {
+                    form.ShowDialog(this);
+                    Reload();
+                    OrdersGridView.FocusedRowHandle = OrdersGridView.LocateByValue("Oid", form.OrderID.Value,
+                        rowHandle => OrdersGridView.FocusedRowHandle = (int)rowHandle);
+                }
+            }
+            private void Reload() {
+                OrdersInstantFeedbackView.Refresh();
+            }
+
+            private void btnNew_ItemClick(object sender, ItemClickEventArgs e) {
+                ShowEditForm(null);
+            }
+
+            private void btnDelete_ItemClick(object sender, ItemClickEventArgs e) {
+                using (Session session = new Session()) {
+                    object orderId = OrdersGridView.GetFocusedRowCellValue(colOid);
+                    Order order = session.GetObjectByKey<Order>(orderId);
+                    session.Delete(order);
+                }
+            }
+        }
+    }
+    ```
+    </details>
+
+* Select the **OrdersInstantFeedbackView** component and double click the **ResolveSession** event to add the event handler. Do the same for the **DismissSession** event
```

Comment:
Why do so many of your sentences not end in a period? One ends in a period, the next does not?

Reply:
```suggestion
* Select the **OrdersInstantFeedbackView** component and double click the **ResolveSession** event to add the event handler. Do the same for the **DismissSession** event.
```

Reply:
da82797c47a2c930446ea01539dfa8c249e5c883